internal/resources/akscluster/schema.go

```diff
@@ -545,12 +605,27 @@ var NodepoolSpecSchema = &schema.Schema{
 				Description: "The OS type of the nodepool",
 				Default:     aksmodel.VmwareTanzuManageV1alpha1AksclusterNodepoolOsTypeLINUX,
 				Optional:    true,
+				ValidateDiagFunc: func(v any, path cty.Path) diag.Diagnostics {
```

Comment:
Nit: I would like this more if the `validateSelectedValue` method returned another method instead of accepting a format string.

Or I just don't like the format string, you can accept the field name and substitute it in a format string in that method. 

But I think returning a method is cleaner here if you want to have a generic validating function.